### PR TITLE
Adding SKLearn OrdinalEncoder as a Transformer

### PR DESCRIPTION
**Pull request by gketronDS**
[please review the [Contribution Guidelines](http://epistasislab.github.io/tpot/contributing/) prior to submitting your pull request. go ahead and delete this line if you've already reviewed said guidelines.]

## What does this PR do?
Adds Scikit-Learn's Ordinal Encoder as a transformer search space component. 

## Where should the reviewer start?
Based on the One Hot Encoder but with the encoder switched out. 

## How should this PR be tested?
I've used it on my experiments so far, and it works as intended. There may be bugs I haven't encountered from swapping onehot encoder for the Ordinal Encoder, so a quick double check couldn't hurt but probably isnt necessary since I runs fine on my copy and the HPC. 

## Any background context you want to provide?
My experiments where running a lot longer from the extra features added by the OneHotEncoder, so I added this to constrain the amount of features I imputed. Useful to prevent possible future neural network components (like a potential GAN or VAE-based imputer) from exploding the runtime. 

## What are the relevant issues?

N/A

## Questions:

- Do the docs need to be updated?
- Does this PR add new (Python) dependencies?

### CODE REVIEW

**Comment by perib** [comment]
in the learned preprocessing example, TPOT could also learn to either select a single ordinal encoding OR one hot encoding (ChoicePipeline) before following with the user/default search space

**Comment by gketronDS** [comment]
I think it would be best to take both encoders out and put them into a preprocessing option like “transformers” or “selectors” so we can use them in sequential pipelines in the future and iterate. Calling one hot imputation in the default preprocessing flag was hard coded and we don’t have a non-hard coded way to do categorical encoding right now other than the onehot transformer object. Idk how much support for learning preprocessing pipelines we want to give but I feel like having it as an option is in line with the overall goal of automation, plus a non-hardcoded option give more leway to take advantage of the new pipeline architectures like choice or sequential.

**Comment by gketronDS** [comment]
From Pedro on Slack: yeah im not sure whether to have a "builtin" learned preprocessing thing, or have it be something the users decide to do. for TPOTClassifier and TPOTRegressor it could be useful and automate an extra few steps. It would be helpful for things like imputation
I was thinking something like
SequentialPipeline(
    [
        #TPOT automatically adds these when needed
        <OneHot or Ordinal Encoding>,
        <Imputers>,
        #Default always included
        <Normal search space>
    ]
)
we can already access individual methods with get_search_space. For example tpot2.config.get_search_space("ColumnOneHotEncoder")

From Me: Since we can just grab from the config anyway. maybe its worth pulling out OneHot from transformers, adding the OrdinalEncoder in for get search space and we can decide if we want make a new preprocessor section for automated preprocessing later? I needed it for my project, so someone else may need that functionality in the future too.

From Pedro: I think both ideas make sense.

**Comment by gketronDS** [comment]
Pedro, are you able to combine these since you made changes to the config recently? Not sure why this is causing a merge error or how to save changes I make.

**Comment by nickotto** [comment]
This code would take away one hot encoding from the transformers and the user would have to manually add column_encoders in their search space. Is this something you agree with?

**Comment by perib** [comment]
perhaps we could benchmark both options to see if it makes a large difference in time/performance/memory?

**Comment by gketronDS** [comment]
Switching back to have encoders in the transformer group per the last TPOT meeting. There was also a merge conflict since the regressors had 'SVR' in the group names. I never changed anything there so I'm not sure why that was giving me an issue. Maybe something to double check. May also need to add the TPOT library disclaimer to column_encoders, and the check is currently failing, so not sure whats going on now.

**Comment by gketronDS** [comment]
Is there a way just to merge [7bc5d21](https://github.com/EpistasisLab/tpot2/pull/157/commits/7bc5d21130a1daaab0f18aec70ed6527d6314573) with the new classifiers regressors and transformers? Trying to do it manually seems to keep failing on test_tpot_estimator_predict which I'm not sure if I even have anything to do with this PR.
